enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+---
+title: installer-telemetry-metrics
+authors:
+  - "@patrickdillon"
+  - "@rna-afk"
+reviewers:
+  - "@abhinavdahiya"
+  - "@brancz"
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@abhinavdahiya"
+  - "@brancz"
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2020-02-07
+last-updated: 2020-02-07
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Installer Telemetry Metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+ > 1. This requires creating an [aggregation pushgateway][weaveworks-aggregation-pushgateway] where the installer would push metrics. How would this be handled and by which team?
+ > 2. If #1 is possible, how would client authentication be administered to prevent [malicious pushing of metrics](risks-and-mitigations)? Authentication based on what? A pull secret? 
+ > 3. How to filter to collect only real user usage and prevent OpenShift developer/Red Hat internal users from skewing metrics? Perhaps filter by version?
+ > 4. Is it possible to determine whether a user modified an ignition config or manifest asset by comparing the asset the installer would generate to the one found on disk?
+ > 5. Are [package-scoped variables](internal-installer-design) acceptable in the proposed installer `metrics` package? 
+
+## Summary
+Telemetry collects metrics from running clusters, but no metrics are obtained from the installer binary.
+This enhancement proposes that the installer would push metrics related to user workflows, installation statistics, and feature usage. The installer would be instrumented to build Prometheus labels, convert them to samples and then push those metrics to an [aggregation pushgateway][weaveworks-aggregation-pushgateway].  
+
+## Motivation
+
+Obtaining metrics from the installer would allow engineers and product managers to understand different user workflows, environments, experiences, and difficulties, as well as an indication of feature usage.  By pushing these metrics to the Telemeter, OpenShift members would be able to create a variety of queries related to these aspects.
+
+### Goals
+
+1. Instrument the installer to collect metrics
+2. Allow the installer to push those metrics to an aggregation pushgateway which is scraped by the Telemetry Prometheus instance
+
+### Non-Goals
+
+1. The metrics collected will be aggregate metrics so metrics will not be relevant to individual installer invocations and will not display detailed data such as log or error messages
```

Comment:
this is a little confusing.

Reply:
I agree, it's confusing for me as well. Does it mean that it's non goal to make it aggregated (reduce cardinality)? I guess the opposite?